### ISSUE TITLE
[HOLD for payment 2023-05-25] [HOLD for payment 2023-05-23] [Feature] Support all file types for attachments

### ISSUE THREAD

**Issue by JmillsExpensify** [Expensify/App#11189]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Try to upload a CSV, XLSX, HEIF or HEFV image. None of them work!

## Expected Result:
We should support all file types, especially since we have an antivirus scanner handling all attachments. 

## Actual Result:
The upload files with the following error message.
<img width="392" alt="image (2)" src="https://user-images.githubusercontent.com/1127863/191621537-efec3c4e-532f-451c-84c7-1f5813434b47.png">

## Workaround:
No workarounds

## Platform:
- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.2.2-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** Everyone
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:** 
**Issue reported by:** @quinthar 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1663790246088889

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kevinksullivan** [comment]
https://www.upwork.com/jobs/~011d5256eaf415611d

**Comment by Santhosh-Sellavel** [comment]
@neil-marcellini 

We have frontend restrictions but could be a backend too as per this comment

https://github.com/Expensify/App/blob/3780e0fe79a96bcfc86e1ac72e12814e5e981633/src/CONST.js#L16-L18

**Comment by youssef-lr** [comment]
So I wasn't able to find anything in the backend (Web-PDF) that only allows certain extensions, when I removed the check for extensions from NewDot I was able to trigger the file upload to S3, however S3 returns this error:

`S3BucketException - Amazon exception [0] - Error executing "PutObject" on "https://dev-receipts-expensify-com.s3.amazonaws.com/w_0574467cb0f96315b5934afbcff08366c947db41.mov"; AWS HTTP error: Client error: 'PUT https://dev-receipts-expensify-com.s3.amazonaws.com/w_0574467cb0f96315b5934afbcff08366c947db41.mov' resulted in a '403 Forbidden'`

I think this might be a matter of changing our S3 configuration to allow all file types.

**Comment by youssef-lr** [comment]
So simply removing the check from the frontend will allow us to upload any file type. I managed to test this on staging, for some reason uploading to S3 from my local machine does not work.

**Comment by JmillsExpensify** [comment]
Sorry guys, created this issue and then bounced. I agree there's a backend component here. Have we created one yet, or are we tracking the E/E part in an existing issue already?

**Comment by JmillsExpensify** [comment]
Actually, I did some sleuthing. Here's the related E/E issue: https://github.com/Expensify/Expensify/issues/145368

**Comment by youssef-lr** [comment]
That's odd, once I removed the code from NewDot that checks for file extensions, I was able just about any file. The code [here](https://github.com/Expensify/Web-Expensify/blob/74bb5a2d30e0b000502549fce3be68f788082fc6/lib/ReportUtils.php#L235-L235) doesn't seem to reject files based on their extension, but rather, it decides whether to display a thumbnail or not. If it's an image or anything that can have a thumbnail, then it returns the thumbnail of it, otherwise, it just displays the filename for download.

**Comment by youssef-lr** [comment]
Ok I think I got it. So if the file satisfies the condition [here](https://github.com/Expensify/Web-Expensify/blob/74bb5a2d30e0b000502549fce3be68f788082fc6/lib/ReportUtils.php#L278), which checks if the file isn't an image/pdf, it can be uploaded no matter its filetype, if it does not, then we check if it has a supported extension [here](https://github.com/Expensify/Web-Expensify/blob/74bb5a2d30e0b000502549fce3be68f788082fc6/lib/Receipt.php#L483).

So TL;DR: In the backend, if the file can't have a thumbnail (zip, video, xls, doc, etc..) we upload it. If the file is an image/pdf: we only allow these filetypes: `/(jpe?g|gif|png|pdf|html?|text|rtf|doc|tiff?|msword|zip|xml|message)/i'`

**Comment by youssef-lr** [comment]
Not overdue.

**Comment by youssef-lr** [comment]
Not overdue, waiting for feedback about how to implement this. Will post in slack to discuss possible solutions.

**Comment by JmillsExpensify** [comment]
>So I think that if we want to allow all files to be uploaded, we need to skip applying the check for filetype for receipts coming from NewDot, but keep it for receipts coming from OldDot, as it wouldn't make sense to allow users to upload any file when we except it to be an actual receipt. Right?

Yes, I think this change is optimizing for sharing attachments on NewDot as attachments. That said, technically you can add attachments to expense report on OldDot as report comment as well (and after all, a chat report is just a report).

**Comment by youssef-lr** [comment]
Got it! I'll start working on the Web-E PR.

**Comment by kevinksullivan** [comment]
Not overdue, I think PR is wip, right @youssef-lr ?

**Comment by youssef-lr** [comment]
Yes @kevinksullivan! Will get to it ASAP.

**Comment by JmillsExpensify** [comment]
Awesome, thanks! This is definitely in line with WhatsApp Quality, so a good one for us to ship, even if it's technically a new feature.

**Comment by youssef-lr** [comment]
Still working on this. Was OOO.

**Comment by JmillsExpensify** [comment]
Going to start profiling this in the weekly WAQ update since I think it's important polish.

**Comment by youssef-lr** [comment]
Posted in engineering-chat for help about the S3 error.

**Comment by youssef-lr** [comment]
PRs ready for review!

**Comment by JmillsExpensify** [comment]
@youssef-lr I wanted to check on the plan moving forward following the regression that you did an awesome job jumping in and fixing. I'd imagine the order of operations is:

 - Get https://github.com/Expensify/Web-Expensify/pull/35361 merged
 - Once that hits production, circle back on the App PR, which looks ready to merge following all Web-E PRs hitting production. 

Is that basically right?

**Comment by JmillsExpensify** [comment]
@youssef-lr You mind helping out with this regression? Everything in this list matters, but in particular, I want to make sure we have added a regression test moving forward.

**Comment by youssef-lr** [comment]
On it @JmillsExpensify.

**Comment by JmillsExpensify** [comment]
> A [regression test](https://expensify.testrail.io/index.php?/dashboard) has been added or updated so that the same bug will not reach production again. Link to the updated test here:

@youssef-lr and i chatted about this one over DM and decided ⁦that this is essentially a one-time change, as we now support all file types and have a very limited list for years. 

More specifically, the issue was that some filetypes have a type called 'octet-stream' which can be just about anything, an image, a PDF, a Word doc, etc.⁦ The was missed during a change for HEIC image, and that broke the check. 

So all in all, I think we are good here and I'm checking that consideration off above.

**Comment by JmillsExpensify** [comment]
>A discussion in #contributor-plus has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

Similar story for this one. We now support all file types, so it's a one time change and should't affect us moving forward.

**Comment by JmillsExpensify** [comment]
I *think* that with this BZ checklist complete and all file types supported in staging right now, we are good to close this one out. Please re-open if I missed something!

**Comment by JmillsExpensify** [comment]
Oh that's the checklist that we just completed above. Ignoring.

**Comment by JmillsExpensify** [comment]
Also as an update, I checked with @youssef-lr via DM. We'll need to circle back to the BZ checklist when the related PRs are complete. A full testing script will also be in order.

**Comment by JmillsExpensify** [comment]
Still working through bugs related to this new feature.

**Comment by JmillsExpensify** [comment]
De-prioritizing this at the moment, in favor of API refactors instead.

**Comment by JmillsExpensify** [comment]
Still de-prioritized for WAQ.

**Comment by JmillsExpensify** [comment]
Not a priority until WAQ is done.

**Comment by JmillsExpensify** [comment]
Still not a priority, though WAQ is ending soon so we should pick this back up not too long from now.

**Comment by roryabraham** [comment]
Let's take this off HOLD?

**Comment by JmillsExpensify** [comment]
Yes, @youssef-lr and I discussed this one before he headed out on vacay. I think the plan is that he'll pick this and related improvements back up when he returns.

**Comment by trjExpensify** [comment]
Why does this have `Feature` in the title but a `Bug` label, as well as a `Monthly` label still. Can someone clean it up?

**Comment by JmillsExpensify** [comment]
Looks like the `bug` label was added more recently. Not sure why, as I just removed it.

**Comment by JmillsExpensify** [comment]
@youssef-lr This is another one where I'm not sure what the next steps following the regression that you mentioned above.

**Comment by JmillsExpensify** [comment]
Commenting on this one to see if that invites our friend Melvin to this issue?

**Comment by youssef-lr** [comment]
@JmillsExpensify given the urgency of this I'm gonna go ahead with the implementation of the solution discussed in this [thread](https://expensify.slack.com/archives/C04QEB4MJEQ/p1683042025870849?thread_ts=1682916637.419009&cid=C04QEB4MJEQ). What do you think?

**Comment by JmillsExpensify** [comment]
Yes, [what Florent posted](https://expensify.slack.com/archives/C04QEB4MJEQ/p1683045629264359?thread_ts=1682916637.419009&cid=C04QEB4MJEQ) at the end of the thread sounds great.

**Comment by JmillsExpensify** [comment]
The timing for this is super tight so I'm going to add the `daily` label back as well.

**Comment by youssef-lr** [comment]
Draft PRs created and it's working nicely for me locally. PRs should be ready for review on Monday!

**Comment by JmillsExpensify** [comment]
@youssef-lr quick question on this issue: Should it still be open?

**Comment by JmillsExpensify** [comment]
Further, @rushatgabhane have been paid you out for this one? I see you reviewed one of the linked PRs.

**Comment by youssef-lr** [comment]
I think we can close!

**Comment by JmillsExpensify** [comment]
Alrighty!

**Comment by JmillsExpensify** [comment]
@rushatgabhane please reach out over New Expensify if anything remains outstanding for you.

**Comment by anmurali** [comment]
I paid Rushat on New Dot

**Comment by JmillsExpensify** [comment]
Thanks!
